provision.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+set -e
+
+readonly dir_backend=employee-rostering-backend
+readonly dir_frontend=employee-rostering-frontend
+
+# Change dir to the project root (where provision.sh is located) to correctly resolve module paths.
+# This needed in case the script was called from a different location than the project root.
+cd "$(dirname "$(readlink -f "$0")")"
+
+command -v oc > /dev/null 2>&1 || {
+  echo >&2 "ERROR: The oc client tool needs to be installed to connect to OpenShift."
+  exit 1
+}
+
+[[ -x $(command -v oc) ]] || {
+  echo >&2 "ERROR: The oc client tool is not executable. Please make it executable by running \
+‘chmod u+x \$(command -v oc)’."
+  exit 1
+}
+
+# Print info about the current user and project
+echo "Current user: $(oc whoami)"
+oc project
+
+if oc status | grep "You have no services" > /dev/null
```

Comment:
Why checking twice (here and on a line 41) for emptiness of the project? Arguably, the method used here might be sensitive to changes. Output of the `oc status` command probably doesn't have to be backward compatible, in contrast to `oc get all -o name`, which should be always empty in case of an empty project.

Reply:
Yes, I think you're right. A slight change in the status output might make the grep check fail and the script would bail out.

The reason I did this is that `oc get all` is somewhat slow (1-3s) so I wanted to use a quicker `oc status` to decide whether project is empty or not for the purpose of printing the initial info.

Good catch, I'll change it to `oc get all` or remove it.